src/goto-programs/goto_convert.cpp

```diff
@@ -1903,7 +1903,7 @@ bool goto_convertt::get_string_constant(
       forall_operands(it, index_op)
         if(it->is_constant())
         {
-          unsigned long i=integer2ulong(
+          const std::size_t i = numeric_cast_v<std::size_t>(
             binary2integer(id2string(to_constant_expr(*it).get_value()), true));
 
```

Comment:
BTW, just ran into that one again (in #3100).

I'd advocate removing this method, given that I don't think it works.

Reply:
Which method are you referring to? I wasn't able to figure out which problem #3100 is solving, so I'd need more info. Thanks!